pkgs/development/python-modules/rapidocr-onnxruntime/default.nix

```diff
@@ -0,0 +1,134 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+
+  fetchurl,
+  linkFarm,
+
+  pythonRelaxDepsHook,
+  pytestCheckHook,
+
+  setuptools,
+  pyclipper,
+  opencv4,
+  numpy,
+  six,
+  shapely,
+  pyyaml,
+  pillow,
+  onnxruntime,
+}:
+let
+  version = "1.3.22";
+
+  src = fetchFromGitHub {
+    owner = "RapidAI";
+    repo = "RapidOCR";
+    rev = "v${version}";
+    hash = "sha256-8h4DFhnI9imr+bYQZdlrl8UKUdpwnGK+SGxLXSMmcag=";
+  };
+
+  modelsRev = "38f67e0558416db6056ddd8613292d23b5ff002c";
+
+  # Please check python/rapidocr_onnxruntime/ch_*/config.yaml for the list of models used in the tests.
+  # The file name should be specified under the key `model_path`.
+  modelFiles = {
+    "PP-OCRv4/ch_PP-OCRv4_det_infer.onnx" = "sha256-0qdyDUWlQlcgix4T42qEeYlMt0FVpe/ilGJRLUL0nak=";
+    "PP-OCRv4/ch_PP-OCRv4_rec_infer.onnx" = "sha256-SPxA8k9tKiB6KxCR00N+s8w+trZ23D75w3OEAFSDaDs=";
+    "PP-OCRv1/ch_ppocr_mobile_v2.0_cls_infer.onnx" = "sha256-5HrO32YyMPiGP/GrDmTdLYK4OPzrWVcUbasYWonWIVw=";
+  };
+
+  models = lib.pipe modelFiles [
+    (lib.mapAttrsToList (
+      name: hash: {
+        inherit name;
+        path = fetchurl {
+          url = "https://huggingface.co/SWHL/RapidOCR/resolve/${modelsRev}/${name}";
+          inherit hash;
+        };
+      }
+    ))
+    (linkFarm "rapidocr-onnxruntime-models")
+  ];
+in
+buildPythonPackage {
+  pname = "rapidocr-onnxruntime";
+  inherit version src;
+  pyproject = true;
+
+  sourceRoot = "${src.name}/python";
+
+  # The setup.py supplied by upstream tries to determine the current version by
+  # fetching the latest version of the package from PyPI, and then bumping the version number.
+  # This is not allowed in the Nix build environment as we do not have internet access,
+  # hence we patch that out and get the version from the build environment directly.
+  patches = [ ./setup-py-override-version-checking.patch ];
+
+  nativeBuildInputs = [ pythonRelaxDepsHook ];
+
+  nativeCheckInputs = [ pytestCheckHook ];
+
+  build-system = [ setuptools ];
+
+  dependencies = [
+    pyclipper
+    opencv4
+    numpy
+    six
+    shapely
+    pyyaml
+    pillow
+    onnxruntime
+  ];
+
+  # Don't use vendored OpenCV
+  pythonRemoveDeps = [ "opencv-python" ];
```

Comment:
`opencv-python` is not a vendor package, it is a standard package called opencv-python in pypi.
We need `pythonRemoveDeps` because we have adopted the `opencv4` as an attribute name.https://pypi.org/project/opencv-python/

Reply:
Fixed.